src/components/07-form/controls/character-count.njk

```diff
@@ -0,0 +1,35 @@
+<form class="usa-form">
+  <div class="usa-character-count" data-module="usa-character-count">
+    <div class="usa-form-group">
+      <label class="usa-label" for="with-hint-input">
+        Text input
+      </label>
+      <span id="with-hint-input-hint" class="usa-hint">
+        This is an input with a character counter.
+      </span>
+      <input class="usa-input usa-character-count__input" id="with-hint-input" maxlength="25" name="with-hint-input" aria-describedby="with-hint-input-info with-hint-input-hint">
+    </div>
+    <span id="with-hint-input-info" class="usa-hint usa-character-count__message" aria-live="polite">
+      You can enter up to 25 characters
+    </span>
+  </div>
+</form>
+
+<form class="usa-form" novalidate>
+  <div class="usa-character-count" data-module="usa-character-count">
+    <div class="usa-form-group">
+      <label class="usa-label" for="with-hint-textarea">
+        Textarea
+      </label>
+      <span id="with-hint-textarea-hint" class="usa-hint">
+        This is a textarea with a character counter.
+      </span>
+      <textarea class="usa-textarea usa-character-count__input" id="with-hint-textarea" maxlength="50" name="with-hint-textarea" rows="5" aria-describedby="with-hint-textarea-info with-hint-textarea-hint"></textarea>
```

Comment:
The following is not a super strongly-held opinion, but thought I'd bring up for discussion.

The `usa-character-count__input` class does not have any CSS properties. It's mainly a JS hook. So wondering if we should use the BEM naming here or use some other convention. Brad Frost discusses class prefixes [his post about css naming architecture](https://bradfrost.com/blog/post/css-architecture-for-design-systems/).  Just trying to think if we cut the number class names the component needs in order to work perhaps.

Another suggestion, if we do keep this convention, is to consider renaming as `usa-character-count__field`, because it's more generic, considering we already have established `usa-input` and `usa-textarea`. Using `__input` on a `<textarea>` feels a little counterintuitive considering that.

Reply:
I like the idea of making it more generic and using `usa-character-count__field`, I'll update that as well.

We could look into extending `usa-input` or `usa-textarea` onto `usa-character-count__field`.

Reply:
@thisisdano would you have an feeling of if/how we should address the issue of needing multiple classes on an element within a component? something like `class="usa-hint usa-character-count__message"`.

Some options for changing the current setup include: 

Styling based off of current USWDS classes and being nested within a `class="usa-character-count"` element being the key selector.

updating the BEM classes to include the styling for `usa-textarea`, `usa-hint`, `usa-input`, etc.. Using `@extend %block-input-general; @extend %block-input-styles;` gets us most of the way there but there would still be more duplication needed.